params/superchain.go

```diff
@@ -75,46 +75,32 @@ func LoadOPStackChainConfig(chainID uint64) (*ChainConfig, error) {
 		TerminalTotalDifficultyPassed: true,
 		Ethash:                        nil,
 		Clique:                        nil,
-		Optimism: &OptimismConfig{
-			EIP1559Elasticity:        6,
-			EIP1559Denominator:       50,
-			EIP1559DenominatorCanyon: newUint64(250),
-		},
+		Optimism:                      nil, // Set below
+	}
+
+	// Optimism parameters are loaded from the genesis
+	gen, err := superchain.LoadGenesis(chainID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load genesis definition for chain %d: %w", chainID, err)
 	}
+	out.Optimism = (*OptimismConfig)(gen.Config.Optimism)
 
 	// special overrides for OP-Stack chains with pre-Regolith upgrade history
 	switch chainID {
-	case OPGoerliChainID:
-		out.LondonBlock = big.NewInt(4061224)
-		out.ArrowGlacierBlock = big.NewInt(4061224)
-		out.GrayGlacierBlock = big.NewInt(4061224)
-		out.MergeNetsplitBlock = big.NewInt(4061224)
-		out.BedrockBlock = big.NewInt(4061224)
-		out.RegolithTime = &OptimismGoerliRegolithTime
-		out.Optimism.EIP1559Elasticity = 10
 	case OPMainnetChainID:
 		out.BerlinBlock = big.NewInt(3950000)
 		out.LondonBlock = big.NewInt(105235063)
 		out.ArrowGlacierBlock = big.NewInt(105235063)
 		out.GrayGlacierBlock = big.NewInt(105235063)
 		out.MergeNetsplitBlock = big.NewInt(105235063)
 		out.BedrockBlock = big.NewInt(105235063)
-	case BaseGoerliChainID:
-		out.RegolithTime = &BaseGoerliRegolithTime
-		out.Optimism.EIP1559Elasticity = 10
-	case baseSepoliaChainID:
-		out.Optimism.EIP1559Elasticity = 10
-	case baseGoerliDevnetChainID:
-		out.RegolithTime = &baseGoerliDevnetRegolithTime
-	case pgnSepoliaChainID:
-		out.Optimism.EIP1559Elasticity = 2
-		out.Optimism.EIP1559Denominator = 8
+		out.Optimism = &OptimismConfig{
+			EIP1559Elasticity:        6,
+			EIP1559Denominator:       50,
+			EIP1559DenominatorCanyon: newUint64(250),
+		}
 	case devnetChainID:
 		out.RegolithTime = &devnetRegolithTime
```

Comment:
This chain is not in the registry, so this override can likely be removed.

Reply:
yes, please